docs/pip.md

```diff
@@ -0,0 +1,162 @@
+  * `https://github.com/containerbuildsystem/operator-manifest/archive/v0.0.3.zip#cachito_hash=sha256:7774db0`
+
+### Recursive dependencies
+
+You will need to include not only the direct dependencies for your application, but all the
+dependencies of those dependencies (recursively) as well. There are two reasons for this. The first
+one is reproducibility (you will need to pin the versions of recursive dependencies as well). The
+second reason is that, to determine the dependencies of your dependencies, Cachito would have to
+execute their `setup.py` files, which is not desirable.
+
+### pip-compile
+
+Good news, there is a tool which does part of the required setup for you. That tool is `pip-compile`
+from the [pip-tools](https://github.com/jazzband/pip-tools) Python package. Below is an example of
+how you would use it.
+
+1. rename your `requirements.txt` to `requirements.in` (convention)
+2. run `pip-compile requirements.in -o requirements.txt`
+
+This is the output of the above command:
+
+```ini
+#
+# This file is autogenerated by pip-compile
+# To update, run:
+#
+#    pip-compile --output-file=requirements.txt requirements.in
+#
+certifi==2020.6.20        # via requests
+chardet==3.0.4            # via requests
+git+https://github.com/containerbuildsystem/dockerfile-parse  # via -r requirements.in
+idna==2.10                # via requests
+https://github.com/containerbuildsystem/operator-manifest/archive/v0.0.3.zip  # via -r requirements.in
+requests==2.24.0          # via -r requirements.in
+ruamel.yaml.clib==0.2.2   # via ruamel.yaml
+ruamel.yaml==0.16.12      # via operator-manifest
+six==1.15.0               # via dockerfile-parse
+urllib3==1.25.10          # via requests
+```
+
+As you can see, `pip-compile` gathered all the recursive dependencies and pinned all PyPI packages.
+It did not pin external dependencies, the mechanism for doing so is specific to Cachito. You can pin
+these beforehand in the `requirements.in` file, but if any of the recursive dependencies are
+external, you may need to edit the generated file anyway.
+
+### Explicit package names
+
+Cachito needs to know the package name for all of your dependencies. For PyPI dependencies, this is
+trivial, as the name is already present in the requirements file. For external dependencies,
+resolving the name may require executing the `setup.py` file. Cachito does have a mechanism for
+extracting package metadata from `setup.py`, but it is very limited. That is why, for external
+dependencies, you will need to explicitly specify package names using one of the mechanisms that
+pip supports.
+
+1. use '@': `dockerfile-parse @ git+https://github.com/containerbuildsystem/dockerfile-parse`
+2. use "#egg": `git+https://github.com/containerbuildsystem/dockerfile-parse#egg=dockerfile-parse`
+
+Note that `pip-compile` seems to ignore the `@` mechanism, so using `#egg` may be preferrable.
```

Comment:
It was not immediately clear to me that you are trying to illustrate two ways to specify the `dockerfile-parse` package name. Can you either present these two examples as two ways to achieve this for a specific package or generalize it with something like `<package-name>`? Maybe don't use inline formatting but use code blocks?

Reply:
Based on this, the examples above in the pinning versions section is incomplete. Can you include a complete example of a `requrements.txt` file that includes all of these referenced parts?

Reply:
Fixed and fixed (and I also used code blocks for the examples in `Pinning versions`). I included an additional example of `requirements.in` rather than the full `requirements.txt`, I hope that is enough